chain/client/src/stateless_validation/state_witness_tracker.rs

```diff
@@ -0,0 +1,201 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+use lru::LruCache;
+use std::hash::Hash;
+
+use crate::metrics;
+use near_async::time::Clock;
+use near_primitives::sharding::ChunkHash;
+use near_primitives::stateless_validation::{ChunkStateWitness, ChunkStateWitnessAck};
+
+/// This is currently used for metrics computation, so we do not need to keep all the
+/// witnesses, so we set some fixed capacity for the witnesses tracked and discard others.
```

Comment:
The documentation strings should generally avoid documenting what uses of the item there are as that can change over time, without the tooling being able to tell you to adjust the comment.

```suggestion
/// Limit to the number of witnesses tracked.
///
/// Other witnesses past this number are discarded (perhaps add a blurb on how.)
```